src/network/mcpe/handler/InGamePacketHandler.php

```diff
@@ -856,6 +860,29 @@ public function handleLabTable(LabTablePacket $packet) : bool{
 		return false; //TODO
 	}
 
+	public function handleLecternUpdate(LecternUpdatePacket $packet) : bool{
+		//Drop book is handled on server side with an interact event.
+		//Total pages is gathered from book.
+		$pos = $packet->blockPosition;
+		$chunkX = $pos->getX() >> Chunk::COORD_BIT_SIZE;
+		$chunkZ = $pos->getZ() >> Chunk::COORD_BIT_SIZE;
+		$world = $this->player->getWorld();
+		if(!$world->isChunkLoaded($chunkX, $chunkZ) || !$world->isChunkGenerated($chunkX, $chunkZ) || $world->isChunkLocked($chunkX, $chunkZ)){
+			return false;
+		}
+		$lectern = $world->getBlockAt($pos->getX(), $pos->getY(), $pos->getZ());
+		if($lectern instanceof Lectern){
+			if($this->player->getPosition()->distance($lectern->getPosition()) <= 15){
+				$book = $lectern->getBook();
+				if($book instanceof WritableBookBase && count($book->getPages()) == $packet->totalPages && $packet->page >= 0 && $packet->page <= $packet->totalPages){
```

Comment:
```suggestion
				if($book !== null && count($book->getPages()) === $packet->totalPages && $packet->page >= 0 && $packet->page < $packet->totalPages){
```

Off-by-one bug here. The page index should not be equal to the page count (the pages are zero-indexed).